### PR TITLE
Docs: Note that PR requires ACCEPTED issue (refs #6568)

### PR DESCRIPTION
**Pull request by pmcelhaney**
**What issue does this pull request address?**
The newly added pull request template states that "all pull requests require an **accepted** issue." However, the contributing docs are no clear on that point. People should have a chance to become aware of this information before they submit the pull request, not after they've done on the work and are about to submit a PR.

**What changes did you make? (Give an overview)**
I updated the docs to clarify that a PR requires an **accepted** issue.

**Is there anything you'd like reviewers to focus on?**
I'm not clear on what makes an issue accepted. I'm assuming a contributor adds an accepted tag to the issue, but the greater question is what are the criteria for having an issue accepted?

### CODE REVIEW

**Comment by pmcelhaney** [comment]
I'll be honest. This PR is motivated by some personal frustration. I'm batting 0/5 in having pull requests accepted, despite having read the contributing guidelines very carefully. Literally just yesterday I sent a PR and found out after the fact that I should not send a PR until the issue is accepted. Because [I love ESLint so much and appreciate the work that's gone into to it](https://twitter.com/patrick_mc/status/722064410064367616) it pains me that every single one of my attempts to give back has been met with what feels like cold resistance. 

Of course, much of this pain is probably self-inflicted. If you've read this far and care to look over my previous pull requests, any insight you can give me into what I may be missing is much appreciated. 

Thanks again for all of the effort you've put into building a quality and useful piece of software.

File: docs/developer-guide/contributing/pull-requests.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ If you'd like to work on a pull request and you've never submitted code before,
 
 1. Sign our [Contributor License Agreement](https://contribute.jquery.org/cla).
 1. Set up a [development environment](../development-environment).
-1. Ensure there's an issue that describes what you're doing. You can create a new issue or just indicate you're [working on an existing issue](working-on-issues).
+1. Ensure there's an issue that describes what you're doing **and the issue has been accepted**. You can create a new issue or just indicate you're [working on an existing issue](working-on-issues).
```
Comment (nzakas) [inline docs/developer-guide/contributing/pull-requests.md]: I don't think this needs to be bold.

**Comment by nzakas** [comment]
@pmcelhaney I'm sorry for the frustration. Looking back over your PRs, it looks like you've frequently started by doing all the work for an idea you have only to end up blocked because there isn't an issue or the issue isn't accepted. I know that can be frustrating, and that's why we ask people to open issues **before** they start work, so we can discuss if we are willing to consider a PR. We want to be mindful of people's time and the best way to do that is to talk before writing any code.

An issue is marked as "accepted" when the team believes it would be a good idea to include in the project. Depending on what the issue is proposing, it can take more or less time to get accepted. Bugs are accepted as soon as a team member can reproduce and agrees that it's a bug, rule enhancements and new rules get accepted when there's a champion plus three +1s, and anything else needs to go through the TSC.

In general, the best way to get started on anything is always to open an issue. You'll usually get some feedback within a couple of days.

**Comment by nzakas** [comment]
Thanks!

**Pull request merged**